.editorconfig

```diff
@@ -168,28 +187,15 @@ dotnet_style_prefer_is_null_check_over_reference_equality_method = true:silent
 #Style - C# 8 features
 csharp_prefer_static_local_function = true:warning
 csharp_prefer_simple_using_statement = true:silent
-csharp_style_prefer_index_operator = false:silent
-csharp_style_prefer_range_operator = false:silent
+csharp_style_prefer_index_operator = true:warning
+csharp_style_prefer_range_operator = true:warning

```

Comment:
Just asking why this is changed [over](https://github.com/ppy/osu/commit/c457571da6d69f41ac7722bda6532fa99722c611) and [over](https://github.com/ppy/osu/pull/13789/files)? So which one is the preferred way of this

Reply:
This is a good point, there have been instances of us asking to revert range syntax usage due to it being unreadable. Probably best to revert.

Reply:
I copied the whole editorconfig from framework repo, to keep consistency.

The severity in editorconfig only affects VS IDE, not command line building. But generally if it's not enforced, setting to slient or suggestion is better.